commit a952abcdaa22116d940ca9cb9253caad1622ae93
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Sep 11 20:51:04 2023 +0000

    auxdisplay: panel: Replace deprecated strncpy() with strtomem_pad()
    
    `strncpy` is deprecated and as such we should prefer more robust and
    less ambiguous interfaces.
    
    In this case, all of `press_str`, `repeat_str` and `release_str` are
    explicitly marked as nonstring:
    |   struct {    /* valid when type == INPUT_TYPE_KBD */
    |           char press_str[sizeof(void *) + sizeof(int)] __nonstring;
    |           char repeat_str[sizeof(void *) + sizeof(int)] __nonstring;
    |           char release_str[sizeof(void *) + sizeof(int)] __nonstring;
    |   } kbd;
    
    ... which makes `strtomem_pad` a suitable replacement as it is
    functionally the same whilst being more obvious about its behavior.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Cc: Kees Cook <keescook@chromium.org>
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Link: https://lore.kernel.org/r/20230911-strncpy-drivers-auxdisplay-panel-c-v1-1-b60bd0ae8552@google.com
    Signed-off-by: Kees Cook <keescook@chromium.org>

diff --git a/drivers/auxdisplay/panel.c b/drivers/auxdisplay/panel.c
index eba04c0de7eb..e20d35bdf5fe 100644
--- a/drivers/auxdisplay/panel.c
+++ b/drivers/auxdisplay/panel.c
@@ -1449,10 +1449,9 @@ static struct logical_input *panel_bind_key(const char *name, const char *press,
 	key->rise_time = 1;
 	key->fall_time = 1;
 
-	strncpy(key->u.kbd.press_str, press, sizeof(key->u.kbd.press_str));
-	strncpy(key->u.kbd.repeat_str, repeat, sizeof(key->u.kbd.repeat_str));
-	strncpy(key->u.kbd.release_str, release,
-		sizeof(key->u.kbd.release_str));
+	strtomem_pad(key->u.kbd.press_str, press, '\0');
+	strtomem_pad(key->u.kbd.repeat_str, repeat, '\0');
+	strtomem_pad(key->u.kbd.release_str, release, '\0');
 	list_add(&key->list, &logical_inputs);
 	return key;
 }